schemas/apps.schema.json

```diff
@@ -0,0 +1,62 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/aiidalab/aiidalab-registry/v1/schemas/apps.schema.json",
+    "$ref": "#/definitions/Welcome",
+    "definitions": {
+        "Welcome": {
+            "type": "object",
+            "propertyNames": {
+              "pattern": "^[A-Za-z][A-Za-z0-9-_]*$"
+            },
+            "additionalProperties": false,
+            "patternProperties": {
+                ".*": {"$ref": "#/definitions/App"}
+            },
+            "title": "Schema for the AiiDAlab apps registry."
+        },
+        "App": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "git_url": {
+                    "type": "string",
+                    "format": "uri",
+                    "qt-uri-protocols": [
+                        "https"
+                    ],
+                    "qt-uri-extensions": [
+                        ".git"
+                    ]
+                },
+                "meta_url": {
+                    "type": "string",
+                    "format": "uri",
+                    "qt-uri-protocols": [
+                        "https"
+                    ],
+                    "qt-uri-extensions": [
+                        ".json"
+                    ]
+                },
+                "categories": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "enum": [
+                          "classical",
+                          "quantum",
+                          "setup",
+                          "tutorials",
+                          "utilities"
```

Comment:
I was wondering, is it possible to take this info directly from the `categories.json`? I am asking this because the content of `categories.json` might change in the future, and I don't want to fix tests for that.

Reply:
This was my idea as well, use a JSON Schema `$ref` here instead, pointing to the `categories.json` schema.

Reply:
No, that is not possible. See [here](https://stackoverflow.com/questions/47454339/use-object-property-keys-as-enum-in-json-schema) for an explanation. The only way this was possible is to redefine `categories.json` to contain a list with the potential values and then a second key that contains the descriptions.

The reason that this is difficult is because "categories.json" (not a schema file) is used as a schema file. I believe the best solution is to define the possible key value pattern for categories (e.g. don't contain spaces or whatever) in `categories.schema.json` and then reference that pattern in `apps.schema.json`. The validation of "is this a valid category" must then be done manually in a separate step.

Reply:
No, that is not possible. See [here](https://stackoverflow.com/questions/47454339/use-object-property-keys-as-enum-in-json-schema) for an explanation. The only way this was possible is to redefine `categories.json` to contain a list with the potential values and then a second key that contains the descriptions.

The reason that this is difficult is because "categories.json" (not a schema file) is used as a schema file. I believe the best solution is to define the possible key value pattern for categories (e.g. don't contain spaces or whatever) in `categories.schema.json` and then reference that pattern in `apps.schema.json`. The validation of "is this a valid category" must then be done manually in a separate step.